renderer/brave_content_renderer_client.cc

```diff
@@ -66,18 +66,43 @@ void BraveContentRendererClient::RenderFrameCreated(
         render_frame, BraveRenderThreadObserver::GetDynamicParams());
   }
 #endif
-
-  new brave_search::BraveSearchRenderFrameObserver(
-      render_frame, content::ISOLATED_WORLD_ID_GLOBAL);
 }
 
 void BraveContentRendererClient::RunScriptsAtDocumentStart(
     content::RenderFrame* render_frame) {
   auto* observer =
       cosmetic_filters::CosmeticFiltersJsRenderFrameObserver::Get(render_frame);
-  // run this before any extensions
+  // Run this before any extensions
   if (observer)
     observer->RunScriptsAtDocumentStart();
 
   ChromeContentRendererClient::RunScriptsAtDocumentStart(render_frame);
 }
+
+void BraveContentRendererClient::WillEvaluateServiceWorkerOnWorkerThread(
+    blink::WebServiceWorkerContextProxy* context_proxy,
+    v8::Local<v8::Context> v8_context,
+    int64_t service_worker_version_id,
+    const GURL& service_worker_scope,
+    const GURL& script_url) {
+  brave_search::BraveSearchSWHolder::GetInstance()
```

Comment:
why isn't this class just owned by BraveContentRendererClient?

Reply:
changed it to be a member